inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/context/ContextManager.java

```diff
@@ -59,4 +62,45 @@ public void attachContextToThread(Thread thread) {
             context.attach();
         }
     }
+
+    @Override
+    public IInspectitContext enterNewContext(boolean copyTagsFromOpenCensus) {
+        val ctx = InspectitContext.createFromCurrent(commonTagsManager, configProvider.getCurrentConfig().getDataProperties(), copyTagsFromOpenCensus);
+        return ContextStackFrame.wrapAndPush(ctx);
+    }
+
+    @Override
+    public IInspectitContext getCurrentContext() {
+        return ContextStackFrame.getCurrent();
+    }
+
+    /**
+     * Will be removed when https://github.com/raphw/byte-buddy/issues/611 is released
+     */
+    private static class ContextStackFrame implements IInspectitContext {
```

Comment:
let's discuss this, please.. lot of time could pass until this is resolved in byte-buddy..